spec/models/nonprofit_spec.rb

```diff
@@ -143,8 +143,27 @@
         expect(nonprofit_with_bad_email_and_website.errors['email'].first).to match /.*invalid.*/ 
       end
 
-      it 'marks website as having errors if they do' do 
-        expect(nonprofit_with_bad_email_and_website.errors['website'].first).to match /.*invalid.*/
+      context 'when the website has errors' do
+        it 'marks as having errors if they do not have a public suffix' do
+          expect(nonprofit_with_bad_email_and_website.errors['website'].first)
+            .to match('is not a valid URL')
+        end
+
+        it 'does not mark website as having errors if they do not have a scheme and adds scheme' do
+          nonprofit_with_bad_email_and_website.update(website: 'a_website.com')
+          expect(nonprofit_with_bad_email_and_website.errors['website'].first)
+            .to be_nil
+          expect(nonprofit_with_bad_email_and_website.website)
+            .to eq('http://a_website.com')
+        end
+
+        it 'marks as having errors and does not replace the scheme if it is different than http or https' do
+          nonprofit_with_bad_email_and_website.update(website: 'ftp://invalid.com')
+            expect(nonprofit_with_bad_email_and_website.errors['website'].first)
+              .to match('is not a valid URL')
+            expect(nonprofit_with_bad_email_and_website.website)
+              .to eq('ftp://invalid.com')
+        end
```

Comment:
Make sure to check the weird situations we need to change the `add_scheme_to_website` for. they include website being:

* an array
* a string with a space
* a number
* a hash

Maybe make a describe section just for that testing that?

Reply:
Or combine the sections here, I don't think it matters a ton.